Druid-Tests/DruidTestRTLCompiler.class.st

```diff
@@ -3165,40 +3141,40 @@ DruidTestRTLCompiler >> gen_primitiveNewWithArg [
 	self CmpCq: 0 R: Extra3Reg.
 	jump5 := self JumpZero: 0.
 	self MoveMw: 32 r: SPReg R: Extra3Reg.
+	self MoveR: Extra3Reg R: SendNumArgsReg.
+	self AddCq: 8 R: SendNumArgsReg.
+	self MoveMw: 32 r: SPReg R: Extra3Reg.
 	self MoveR: Extra3Reg R: Extra2Reg.
 	self AddCq: 8 R: Extra2Reg.
-	self MoveMw: 32 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: TempReg.
-	self AddCq: 8 R: TempReg.
 	self MoveCq: 8 R: Extra1Reg.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self MulR: Extra3Reg R: Extra1Reg.
-	self AddR: Extra1Reg R: TempReg.
-	self SubCq: 1 R: TempReg.
-	self MoveR: TempReg R: Extra1Reg.
+	self AddR: Extra1Reg R: Extra2Reg.
+	self SubCq: 1 R: Extra2Reg.
```

Comment:
Strange why here the 8 was not propagated.

Extra1Reg contains a constant.
Then this could have been

AddCq: 8 R: Extra2Reg

and with some extra love that would mean

Add: 8 R: X
Sub: 1 R: X
=>
Add: 7 R: X

right?